### ISSUE TITLE
[typhoon] Update to v2.1.0 broke my site

### ISSUE THREAD

**Issue by IvanMontillaM** [getgrav/grav-premium-issues#118]
Good evening,

I updated Typhoon theme from v2.0.1 to v.2.1.0 and the website stopped working suddenly.

```
Server Error
Sorry, something went terribly wrong!
0 - An exception has been thrown during the rendering of a template ("Identifier "langswitcher" is not defined.").
For further details please review your logs/ folder, or enable displaying of errors in your system configuration.
```

I didn't have `langswitcher` plugin previously installed as I just write in English and I don't bother with any other language. I installed it thinking this might resolve the issue, but it didn't. I installed it via `bin/gpm install langswitcher`.

I tried the following command `bin/gpm install typhoon`, overwrote accordingly and then `bin/grav clean && bin/grav cache` just in case and still this didn't solve.

What I did was to comment out lines `125` through `184` in `./typhoon.php` and this did the trick, albeit I suspect this is not the cleanest hack.

If you want to reproduce my build, I just used: [Blog Skeleton](https://github.com/trilbymedia/grav-skeleton-typhoon-blog)

Body of the log:
```
[2021-06-25 21:36:12] grav.CRITICAL: An exception has been thrown during the rendering of a template ("Identifier "langswitcher" is not defined."). - Trace: 
#0 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(420): Twig\Template->displayWithErrorHandling(Array, Array) 
#1 /home/ivanmont/public_html/cache/twig/27/27646eb16de049cec96a4a47e2043df0ba4b6ecac0d614bcd4ed6c0a3e99b7b4.php(76): Twig\Template->display(Array) 
#2 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(453): __TwigTemplate_80e5ce4a8cb03b1ffee2d8e738a30d53ecf8007219709e03300d0524297c6752->doDisplay(Array, Array) 
#3 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(420): Twig\Template->displayWithErrorHandling(Array, Array) 
#4 /home/ivanmont/public_html/cache/twig/2f/2f149dfa2c162b2410cde72132123d030c433b34ab9f45cc1891a7ea49ea0361.php(249): Twig\Template->display(Array) 
#5 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(214): __TwigTemplate_50c1ec9b278e91278f9eb3aea22df2663589293133b15a83c294bc1409aedca2->block_header(Array, Array) 
#6 /home/ivanmont/public_html/cache/twig/2f/2f149dfa2c162b2410cde72132123d030c433b34ab9f45cc1891a7ea49ea0361.php(120): Twig\Template->displayBlock('header', Array, Array) 
#7 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(214): __TwigTemplate_50c1ec9b278e91278f9eb3aea22df2663589293133b15a83c294bc1409aedca2->block_render_body(Array, Array) 
#8 /home/ivanmont/public_html/cache/twig/ef/efc5a0b3d284523b3d4f085b960bb769c20e7eabe1b11bae2be44531cfce7f54.php(55): Twig\Template->displayBlock('render_body', Array, Array) 
#9 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(453): __TwigTemplate_247abaeea009c7f9f5b630722d46771658ef3730dc08810368d751d24590d197->doDisplay(Array, Array) 
#10 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(420): Twig\Template->displayWithErrorHandling(Array, Array) 
#11 /home/ivanmont/public_html/cache/twig/2f/2f149dfa2c162b2410cde72132123d030c433b34ab9f45cc1891a7ea49ea0361.php(57): Twig\Template->display(Array, Array) 
#12 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(453): __TwigTemplate_50c1ec9b278e91278f9eb3aea22df2663589293133b15a83c294bc1409aedca2->doDisplay(Array, Array) 
#13 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(420): Twig\Template->displayWithErrorHandling(Array, Array) 
#14 /home/ivanmont/public_html/cache/twig/ce/ce2f8f162393f18f84b6e124ddbc755e1e489c7435bf8786e7520fbb0e0cfc6f.php(47): Twig\Template->display(Array, Array) 
#15 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(453): __TwigTemplate_fffc757f1f9f650a1c10dd4324ef92321e06f880b8033cf5dd64ae340b413273->doDisplay(Array, Array) 
#16 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(420): Twig\Template->displayWithErrorHandling(Array, Array) 
#17 /home/ivanmont/public_html/vendor/twig/twig/src/Template.php(432): Twig\Template->display(Array) 
#18 /home/ivanmont/public_html/vendor/twig/twig/src/TemplateWrapper.php(47): Twig\Template->render(Array, Array) 
#19 /home/ivanmont/public_html/vendor/twig/twig/src/Environment.php(384): Twig\TemplateWrapper->render(Array) 
#20 /home/ivanmont/public_html/system/src/Grav/Common/Twig/Twig.php(435): Twig\Environment->render('blog.html.twig', Array) 
#21 /home/ivanmont/public_html/system/src/Grav/Common/Service/OutputServiceProvider.php(36): Grav\Common\Twig\Twig->processSite('html') 
#22 /home/ivanmont/public_html/vendor/pimple/pimple/src/Pimple/Container.php(118): Grav\Common\Service\OutputServiceProvider->Grav\Common\Service\{closure}(Object(Grav\Common\Grav)) 
#23 /home/ivanmont/public_html/system/src/Grav/Common/Processors/RenderProcessor.php(40): Pimple\Container->offsetGet('output') 
#24 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\RenderProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#25 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#26 /home/ivanmont/public_html/system/src/Grav/Common/Processors/DebuggerAssetsProcessor.php(38): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#27 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\DebuggerAssetsProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#28 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#29 /home/ivanmont/public_html/system/src/Grav/Common/Processors/PagesProcessor.php(81): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#30 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\PagesProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#31 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#32 /home/ivanmont/public_html/system/src/Grav/Common/Processors/TwigProcessor.php(38): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#33 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\TwigProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#34 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#35 /home/ivanmont/public_html/system/src/Grav/Common/Processors/AssetsProcessor.php(39): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#36 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\AssetsProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#37 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#38 /home/ivanmont/public_html/system/src/Grav/Common/Processors/SchedulerProcessor.php(40): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#39 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\SchedulerProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#40 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#41 /home/ivanmont/public_html/system/src/Grav/Common/Processors/BackupsProcessor.php(39): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#42 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\BackupsProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#43 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#44 /home/ivanmont/public_html/system/src/Grav/Common/Processors/TasksProcessor.php(69): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#45 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\TasksProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#46 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#47 /home/ivanmont/public_html/system/src/Grav/Common/Processors/RequestProcessor.php(63): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#48 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\RequestProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#49 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#50 /home/ivanmont/public_html/system/src/Grav/Common/Processors/ThemesProcessor.php(38): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#51 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\ThemesProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#52 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#53 /home/ivanmont/public_html/system/src/Grav/Common/Processors/PluginsProcessor.php(39): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#54 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\PluginsProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#55 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#56 /home/ivanmont/public_html/system/src/Grav/Common/Processors/InitializeProcessor.php(130): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#57 /home/ivanmont/public_html/system/src/Grav/Common/Debugger.php(546): Grav\Common\Processors\InitializeProcessor::Grav\Common\Processors\{closure}() 
#58 /home/ivanmont/public_html/system/src/Grav/Common/Processors/InitializeProcessor.php(131): Grav\Common\Debugger->profile(Object(Closure)) 
#59 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(50): Grav\Common\Processors\InitializeProcessor->process(Object(Nyholm\Psr7\ServerRequest), Object(Grav\Framework\RequestHandler\RequestHandler)) 
#60 /home/ivanmont/public_html/system/src/Grav/Framework/RequestHandler/Traits/RequestHandlerTrait.php(62): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#61 /home/ivanmont/public_html/system/src/Grav/Common/Grav.php(295): Grav\Framework\RequestHandler\RequestHandler->handle(Object(Nyholm\Psr7\ServerRequest)) 
#62 /home/ivanmont/public_html/index.php(58): Grav\Common\Grav->process() 
#63 {main} [] []
```
List of plugins I have:
```
admin
archives
breadcrumbs
color-tools
cookieconsent
email
error
feed
flex-objects
form
langswitcher
license-manager
login
markdown-notices
pagination
problems
random
relatedpages
seo
shortcode-core
simplesearch
sitemap
svg-icons
taxonomylist
```

Keep it up with the great theme!

**Comment by rhukster** [comment]
Ah sorry about this.  I just need to check for the existence of the langswitcher plugin. I’ll get a fix out ASAP.

**Comment by rhukster** [comment]
I've just released a 2.1.1 with a fix..should be available soon.  

Quick fix is to change lines 124-125 in typhoon.php:

```php
        $langswitcher = $this->grav['langswitcher'] ?? null;
        if ($langswitcher && $language->enabled() && $this->config->get( 'theme.menu.langswitcher.enabled')) {
```

**Comment by IvanMontillaM** [comment]
Dear @rhukster, thank you very much for the fix! Have a nice weekend! :-)
